### DISCUSSION TITLE
Can we use this library in traditional applications?

### DISCUSSION THREAD

**Discussion by mohammadobaid1** [Consensys/gnark#107] [General]
We are thinking about implementing zero-knowledge proof on sensitive data streams so that data providers don't need to expose their data and verifier can also verify data. So can we use this library in the traditional application or it is built specifically for blockchain base usecase?

**Comment by gbotrel** [comment]
Hello -- the library is independent from the blockchain layer, and can be used in a non-blockchain context. 

That being said, some design tradeoffs may favor blockchain-applications; no trusted setup (decentralization / trust issue), small proofs, short verifier time, etc.

**Reply by quevas13** [reply]
Hello,

As far as I know gnark uses Groth16 or PLONK, so there is always a trusted setup. Am I right?

Thank you

**Reply by gbotrel** [reply]
Yes correct; more info on the docs: https://docs.gnark.consensys.net/en/latest/Concepts/schemes_curves/ . 
Groth16 uses a trusted setup per circuit. 
PlonK can be instantiated with different commitment schemes. At the moment we support KZG, which needs a trusted setup (upgradable). Support for PlonK-FRI (no trusted setup) is on the roadmap.

**Comment by 311923** [comment]
Any update on when PlonK-FRI will be supported?

**Reply by ivokub** [reply]
We do have PLONK-FRI as an experimental backend. See https://github.com/Consensys/gnark/tree/master/backend/plonkfri

However, it is still highly experimental and fails for some circuits. We are not planning to fix or improve it for now as there doesn't seem to be a practical use case for it. We may even remove the support in the future for simplifying the implementations.
